src/toxcore/dht/codec.rs

```diff
@@ -290,8 +423,10 @@ mod tests {
         let mut buf = BytesMut::new();
 
         buf.extend_from_slice(b"\xFF");
+
+        let res = codec.decode(&mut buf);
         // not enought bytes to decode EncryptedPacket
-        assert!(codec.decode(&mut buf).is_err());
+        assert!(res.is_err());
```

Comment:
Check for exact error?

Reply:
Can't derive PartialEq bacause std::io::Error doesn't implement PartialEq

Reply:
PartialEq is added.